shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionReducer.kt

```diff
@@ -138,4 +143,19 @@ class StepCompletionReducer(private val stepRoute: StepRoute) : StateReducer<Sta
                 state to setOf(Action.LogAnalyticEvent(event))
             }
         } ?: (state to emptySet())
+
+    private fun getNextStepRouteForLearningActivity(learningActivity: LearningActivity?): StepRoute? {
+        if (learningActivity == null) {
+            return null
+        }
+
+        return if (learningActivity.type == LearningActivityType.LEARN_TOPIC &&
+            learningActivity.targetId != null &&
+            learningActivity.targetType == LearningActivityTargetType.STEP
```

Comment:
It's better to extract this complex condition into a variable with explaining name. Generally I suggest replacing 2 separate ifs with a single when.